pyglet/math.py

```diff
@@ -52,69 +52,263 @@ def clamp(num, min_val, max_val):
+        """Reverse add. Required for functionality with sum()
+        """
+        if other == 0:
+            return self
+        else:
+            return self.__add__(other)
+
+    def from_magnitude(self, magnitude):
+        """Create a new Vector of the given magnitude by normalizing, then scaling the vector. The heading remains unchanged.
+
+        :parameters:
+            `magnitude` : int or float :
+                The magnitude of the new vector.
+
+        :returns: A new vector with the magnitude.
+        :rtype: Vec2
+        """
+        return self.normalize().scale(magnitude)
+
+    def from_heading(self, heading):
+        """Create a new vector of the same magnitude with the given heading. I.e. Rotate the vector to the heading.
+
+        :parameters:
+            `heading` : int or float :
+                The angle of the new vector in radians.
+
+        :returns: A new vector with the given heading.
+        :rtype: Vec2
+        """
+        mag = self.__abs__()
+        return self.__class__(mag * _math.cos(heading), mag * _math.sin(heading))
+
+    def limit(self, max):
+        """Limit the magnitude of the vector to the value used for the max parameter.
+
+        :parameters:
+            `max`  : int or float :
+                The maximum magnitude for the vector.
+
+        :returns: Either self or a new vector with the maximum magnitude.
+        :rtype: Vec2
+        """
+        if self[0] ** 2 + self[1] ** 2 > max * max:
+            return self.from_magnitude(max)
+        return self
 
     def lerp(self, other, alpha):
-        return Vec2(self[0] + (alpha * (other[0] - self[0])),
-                    self[1] + (alpha * (other[1] - self[1])))
+        """Create a new vector lineraly interpolated between this vector and another vector.
+
+        :parameters:
+            `other`  : Vec2 :
+                The vector to be linerly interpolated to.
+            `alpha` : float or int :
+                The amount of interpolation.
+                Some value between 0.0 (this vector) and 1.0 (other vector).
+                0.5 is halfway inbetween.
+
+        :returns: A new interpolated vector.
+        :rtype: Vec2
+        """
+        return self.__class__(self[0] + (alpha * (other[0] - self[0])),
+                              self[1] + (alpha * (other[1] - self[1])))
 
     def scale(self, value):
-        return Vec2(self[0] * value, self[1] * value)
+        """Multiply the vector by a scalar value.
+
+        :parameters:
+            `value`  : int or float :
+                The ammount to be scaled by
+
+        :returns: A new vector scaled by the value.
+        :rtype: Vec2
+        """
+        return self.__class__(self[0] * value, self[1] * value)
+
+    def rotate(self, angle):
+        """Create a new Vector rotated by the angle. The magnitude remains unchanged.
+
+        :parameters:
+            `angle` : int or float :
+                The angle to rotate by
```

Comment:
Is this radians or degrees?

Reply:
no clue, it's from the original Vec2 code...

Reply:
I think it will be radians, might be worth adding that to the docstring

Reply:
i’ll add that

Reply:
@LennyPhoenix Done!

Reply:
I think "radiants" is a typo, should be all clear once that's fixed!

Reply:
Done